fe/fe-core/src/main/java/com/starrocks/load/ExportJob.java

```diff
@@ -425,6 +426,54 @@ private void genCoordinators(ExportStmt stmt, List<PlanFragment> fragments, List
         LOG.info("create {} coordintors for export job: {}", coordList.size(), id);
     }
 
+    // For olap table, it may have multiple replica, 
+    // rebalance process may schedule tablet from one BE to another BE.
+    // In such case, coord will return 'Not found tablet xxx' error. To solve this, 
+    // we need to find a new replica for that tablet and generate a new coord.
+    // Also, if the version has been compacted in one BE's tablet, coord will return 
+    // 'version already been compacted' error msg, find a new replica may be able to 
+    // alleviate this problem.
+    public Coordinator resetCoord(int taskIndex, UUID uuid) throws UserException {
+        Coordinator coord = coordList.get(taskIndex);
+        OlapScanNode olapScanNode = (OlapScanNode) coord.getScanNodes().get(0);
+        List<TScanRangeLocations> locations = olapScanNode.getScanRangeLocations(0);
+        if (locations.size() == 0) {
+            throw new UserException("SubExportTask " + taskIndex + " scan range is empty");
+        }
+
+        OlapScanNode newOlapScanNode = new OlapScanNode(new PlanNodeId(0), exportTupleDesc, "OlapScanNodeForExport");
+        Analyzer tmpAnalyzer = new Analyzer(GlobalStateMgr.getCurrentState(), null);
+        newOlapScanNode.setColumnFilters(Maps.newHashMap());
+        newOlapScanNode.setIsPreAggregation(false, "This an export operation");
+        newOlapScanNode.setCanTurnOnPreAggr(false);
+        newOlapScanNode.init(tmpAnalyzer);
+        newOlapScanNode.selectBestRollupByRollupSelector();
+        List<TScanRangeLocations> newLocations = newOlapScanNode.updateScanRangeLocations(locations);
+
+        // random select a new location for each TScanRangeLocations
+        for (TScanRangeLocations tablet : newLocations) {
+            List<TScanRangeLocation> tabletLocations = tablet.getLocations();
+            Collections.shuffle(tabletLocations);
+            tablet.setLocations(tabletLocations.subList(0, 1));
+        }
+
+        OlapScanNode newTaskScanNode = genOlapScanNodeByLocation(newLocations);
+        PlanFragment newFragment = genPlanFragment(exportTable.getType(), newTaskScanNode, taskIndex);
+
+        TUniqueId queryId = new TUniqueId(coord.getQueryId().hi, uuid.getLeastSignificantBits());
+        Coordinator newCoord = new Coordinator(
+                id, queryId, desc, Lists.newArrayList(newFragment), Lists.newArrayList(newTaskScanNode),
+                TimeUtils.DEFAULT_TIME_ZONE, coord.getStartTime(), Maps.newHashMap());
```

Comment:
can you update the scan range in the old coordinator instead of generating a new coordinator?

Reply:
This approve is more complex than current approach. We need coord to assign locations to differnt BE according to BackendSelector.